app/src/main/java/ru/netology/fmhandroid/adapter/PatientListAdapter.kt

```diff
@@ -40,12 +41,20 @@ class PatientListAdapter(
         fun bind(patient: Patient) {
             binding.apply {
                 (patient.lastName + " " + patient.firstName + " " + patient.middleName).also { patientName.text = it }
-//                patientLocation.text = if (patient.status == PatientStatusEnum.ACTIVE) "Да" else "Нет"
+                patientLocation.text = selectionOfPatientStatus(patient).toString()
                 patientName.setOnClickListener {
                     onInterractionListener.onOpenCard(patient)
                 }
             }
         }
+
+        private fun selectionOfPatientStatus(patient: Patient) =
+            when (patient.status) {
+                PatientStatusEnum.ACTIVE -> R.string.patient_status_active
+                PatientStatusEnum.EXPECTED -> R.string.patient_status_expected
+                PatientStatusEnum.DISCHARGED -> R.string.patient_status_discharged
+                null -> "-"
+            }
```

Comment:
Может что-то такое попробовать?
```suggestion
        private val PatientStatusEnum.textResId get() = when (this) {
            PatientStatusEnum.ACTIVE -> R.string.patient_status_active
            PatientStatusEnum.EXPECTED -> R.string.patient_status_expected
            PatientStatusEnum.DISCHARGED -> R.string.patient_status_discharged
        }
```

Reply:
Красивое решение, спасибо
Исправлено